### PR TITLE
Fix deadlock from `bf_read`

### PR DESCRIPTION
**Pull request by rdaum**
Didn't fully diagnose why, but the scheduler `dashmap` tasks map was deadlocking when returning from a read.

Switching back to a mutex'd hashmap fixes the problem.

### CODE REVIEW

File: crates/kernel/src/tasks/scheduler.rs [hunk 1]
```diff
@@ -406,19 +411,24 @@ impl Scheduler {
...
-        for t in self.tasks.iter() {
-            let task = t.value();
-            let tcs = task.task_control_sender.clone();
-            if let Err(e) = tcs.send(TaskControlMsg::Abort) {
-                warn!(task_id = task.task_id, error = ?e, "Could not send abort for task. Already dead?");
-                continue;
+        {
+            // Send shut down to all the tasks.
+            let tasks = self.tasks.lock().unwrap();
+            for (_, task) in tasks.iter() {
```
Comment (abesto) [inline crates/kernel/src/tasks/scheduler.rs:417]: Alternatively: `tasks.iter_values()`

**Review by abesto — APPROVED** [review]

File: crates/kernel/src/tasks/scheduler.rs [hunk 2]
```diff
@@ -678,10 +697,21 @@ impl Scheduler {
...
                 {
-                    let Some(mut task) = self.tasks.get_mut(&task_id) else {
+                    let mut tasks = self.tasks.lock().unwrap();
+                    let Some(task) = tasks.get_mut(&task_id) else {
                         warn!(task_id, "Task not found for input request");
                         return None;
                     };
+                    // Commit the session (not DB transaction) to make sure current output is
+                    // flushed up to the prompt point.
+                    let Ok(()) = task.session.commit() else {
```
Comment (abesto) [inline crates/kernel/src/tasks/scheduler.rs:707]: I *think* this is unrelated?

Comment (rdaum) [reply crates/kernel/src/tasks/scheduler.rs:705]: Actually intentional, just didn't mention in commit
